components/ai_chat/resources/page/components/markdown_renderer/index.tsx

```diff
@@ -0,0 +1,108 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import Markdown from 'react-markdown'
+
+import styles from './style.module.scss'
+import CaretSVG from '../svg/caret'
+
+const CodeBlock = React.lazy(async () => ({
+  default: (await import('../code_block')).default.Block
+}))
+const CodeInline = React.lazy(async () => ({
+  default: (await import('../code_block')).default.Inline
+}))
+
+const allowedElements = [
+  // Headings
+  'h1',
+  'h2',
+  'h3',
+  'h4',
```

Comment:
So we're not allowing some header levels, but also not attempting to prevent the llm from including them. So it's possible we'll present the odd ugly `##### Title` to the user? Perhaps we can just include them here, just in case, and style them any way that is better than markdown code.

Reply:
ok, added back h5,h6 and styled suggested by @aguscruiz